### PR TITLE
Multifix

### PR DESCRIPTION
**Pull request by dickermoshe**
Fix https://github.com/Carapacik/swagger_parser/issues/192
Fix https://github.com/Carapacik/swagger_parser/issues/190
Fix https://github.com/Carapacik/swagger_parser/issues/195

### CODE REVIEW

**Comment by dickermoshe** [comment]
Other fixes will require much more work. These were pretty trivial.:

value is renamed to value_ to account for the retrofit bug.
$ is escaped
Custom types are converted to Pascal

**Comment by Carapacik** [comment]
@dickermoshe plz resolve conflicts

**Comment by dickermoshe** [comment]
Done

File: swagger_parser/lib/src/utils/type_utils.dart [hunk 1]
```diff
@@ -254,3 +257,11 @@ final _nameRegExp = RegExp(r'^[a-zA-Z_][a-zA-Z\d_]*$');
     },
   );
 }
+
+// Protect json key from incorrect symbols, keywords, etc.
```
Comment (Carapacik) [inline swagger_parser/lib/src/utils/type_utils.dart]:
```suggestion
/// Protect json key from incorrect symbols, keywords, etc.
```

File: swagger_parser/lib/src/parser/utils/type_utils.dart [hunk 2]
```diff
@@ -213,3 +213,7 @@ final _nameRegExp = RegExp(r'^[a-zA-Z_][a-zA-Z\d_]*$');
     },
   );
 }
+
```
Comment (Carapacik) [inline swagger_parser/lib/src/parser/utils/type_utils.dart:216]: Add docs like in functions above
```
/// Protect name from incorrect symbols, keywords, etc.
```

**Review by Carapacik — CHANGES_REQUESTED** [review]

File: swagger_parser/lib/src/parser/utils/type_utils.dart (hunk 2 above)
Comment (dickermoshe) [reply swagger_parser/lib/src/parser/utils/type_utils.dart:216]: It had one before the merge conflict 😊
Done

**Review by Carapacik — APPROVED** [review]

**Comment by Carapacik** [comment]
LGTM

**Pull request merged**